### DISCUSSION TITLE
Create fork of numeral at getredash/Numeral-js

### DISCUSSION THREAD

**Discussion by eradman** [getredash/redash#6731] [Ideas]
The last release of numeral.js has a major flaw, which is that it cannot parse small numbers

```
> var numeral = require("numeral");
> numeral('1.2e-9').format('0.00000000000')
'NaN'
```

Several fixes have been proposed, such as https://github.com/adamwdraper/Numeral-js/pull/564, but none reviewed.

We tried switching to numbro in https://github.com/getredash/redash/pull/6344 but had to revert it because integers are not rounded properly.

Elastic has a [public fork of numeral.js](https://github.com/elastic/numeral-js/releases/tag/v3.0.0), but it is not ideal since this fork does not have `numeral.options.scalePercentBy100`, which is a feature Redash uses.

We should consider creating a fork of numeral.js (as was done with [sql-formatter](https://github.com/getredash/sql-formatter)) so that we can apply bug fixes to this library.

**Comment by eradman** [comment]
This problem continued to plague us, we reluctantly forked Numeral so that we could apply a fix

```
yarn add 'numeral@https://github.com/StarfishStorage/Numeral-js'
cd viz-lib
yarn add 'numeral@https://github.com/StarfishStorage/Numeral-js'
cd
```

**Reply by justinclift** [reply]
Cool.  It's such a pain that we have to do this.  But at least we're getting it solved. :smile:
